### PR TITLE
Clean up deprecation warnings

### PR DESCRIPTION
**Pull request by Davidy22**
The deprecation spring cleaning mentioned in #19371

Notes:
Changed some strings in tests to raw strings, which will change the literal content of the strings as they are fed into whatever machine handles them.
Test cases for past in the past/past_key_values switch changed/removed due to warning of impending removal
Most of the warnings defined and thrown by transformers functions were left alone

# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Wow, thanks a lot for all this work! As this touches many many files, I'd love to have other pair of eyes on this, @amyeroberts and @LysandreJik for a general review and @gante or @Rocketknight1 for the past/past_key_values deprecation in the TF side.

**Comment by amyeroberts** [comment]
@Davidy22 Thanks for opening this PR and all of the fixes! 💪 

Just taking a look now. It would really helpful as a reviewer, and for future reference if you could give a quick summary of the deprecation warnings you tackled.

As a quick note regarding `PIL.Image.XXX` -> `PIL.Image.Resampling.XXX`, these changes might be breaking (c.f. [open transformers issue](https://github.com/huggingface/transformers/issues/19569) and [related PR in diffusers](https://github.com/huggingface/diffusers/pull/788) as it requires `Pillow>=9.1.0` which is not currently enforced. @sgugger What is the typical strategy for handling dependancy version updates?

**Comment by sgugger** [comment]
Thanks for pointing this out Amy! In this case, Pillow 9.1.0 is too recent to be pinned as a minimum version (our general rule of thumb is to provide a support for 2 years and this is only from April). So we will need to move those objects to our `image_utils.py` where we can do some checks like:
```py
if version.parse(version.parse(PIL.__version__).base_version) >= version.parse("9.1.0"):
    # Define enum using `PIL.Image.Resampling`
else:
    # Define enum using `PIL.Image`
```

As for the actual enum, we could name it `PILImageResampling`?

**Comment by Davidy22** [comment]
Oh whoops I probably should have taken a couple more notes, I only wrote down a couple of things that stood out as things that'd functionally change something. Summary from re-skimming through the listed changes:

np types swapped with equivalent python default types
PIL Resampling options switched to the new recommended location in the PIL library
Some strings with non-python escapes changed to r strings.
dict_type -> dict parameter in dictionary.Dictionary
Usages of past changed to past_key_values
topk -> top_k

**Review by Rocketknight1 — COMMENTED** [review]
I'm okay with renaming `past` to `past_key_values` in GPT-2/GPT-J/XGLM but I'm going to wait for confirmation from @gante because this might affect the TF generation code!

File: src/transformers/models/gpt2/modeling_tf_gpt2.py
Comment (gante) [reply src/transformers/models/gpt2/modeling_tf_gpt2.py:357]: We gracefully handle both `past` and `past_key_values` in the [forward pass input preparation](https://github.com/huggingface/transformers/blob/90071fe42b8f4bcb560c70a9c2f0a2d0deadebb2/src/transformers/modeling_tf_utils.py#L464) and in [generate's past handling](https://github.com/huggingface/transformers/blob/90071fe42b8f4bcb560c70a9c2f0a2d0deadebb2/src/transformers/generation_tf_utils.py#L1958). So we can update the old cases in their forward passes 🙌 

There is an exception: In the models' `prepare_inputs_for_generation` arguments, we expect `past` and not `past_key_values` ([example](https://github.com/huggingface/transformers/blob/90071fe42b8f4bcb560c70a9c2f0a2d0deadebb2/src/transformers/models/whisper/modeling_tf_whisper.py#L1359)). It seems that part isn't touched in the PR, so it's all good 😎

**Review by gante — APPROVED** [review]
Thank you for updating so many pieces of deprecated code ❤️

**Review by amyeroberts — APPROVED** [review]
Thanks for all of these fixes 🔧 

All LGTM! Could you implement the PILResampling enum @sgugger suggested? Once that's done I think we'll be good to merge 👍

File: src/transformers/models/gpt2/modeling_tf_gpt2.py
Comment (Davidy22) [reply src/transformers/models/gpt2/modeling_tf_gpt2.py:357]: From what I gathered. there would only be exceptions thrown from anything decorated with `@unpack_inputs`, so I left the methods that had arguments named past that didn't have the decorator alone. I did change the return values of some of them though, swapping "past" dict keys with "past_key_values"

File: src/transformers/image_utils.py
Comment (sgugger) [inline src/transformers/image_utils.py:369]: This one needs to be changed to `PILImageResampling`

**Review by sgugger — APPROVED** [review]

**Comment by Davidy22** [comment]
Added PILImageResampling, dealt with some funky import issues in one file that weren't happening in any of the other files, don't know why specifically the flava test file would have issues importing from image_utils, hoping it's not some thing that actually also happens in other files but doesn't get surfaced because it's not covered or something

**Review by LysandreJik — APPROVED** [review]
Love this PR, very impressive work! Thank you @Davidy22!

**Pull request merged**
